dracut/30ignition/flatcar-metadata-hostname.service

```diff
@@ -8,6 +8,33 @@ Wants=systemd-networkd.service initrd-root-fs.target
 # Ensure Ignition can overwrite /etc/hostname
 Before=ignition-files.service
 
+# From coreos/afterburn:dracut/30afterburn/afterburn-hostname.service
+# (but with coreos.oem.id/flatcar.oem.id):
+# These platforms do not provide the hostname via DHCP
+# options, thus it needs to be fetched from the metadata
+# and statically applied on first-boot.
+ConditionKernelCommandLine=|coreos.oem.id=aliyun
+ConditionKernelCommandLine=|flatcar.oem.id=aliyun
+ConditionKernelCommandLine=|coreos.oem.id=azure
+ConditionKernelCommandLine=|flatcar.oem.id=azure
+ConditionKernelCommandLine=|coreos.oem.id=azurestack
+ConditionKernelCommandLine=|flatcar.oem.id=azurestack
+ConditionKernelCommandLine=|coreos.oem.id=digitalocean
+ConditionKernelCommandLine=|flatcar.oem.id=digitalocean
+ConditionKernelCommandLine=|coreos.oem.id=exoscale
+ConditionKernelCommandLine=|flatcar.oem.id=exoscale
+ConditionKernelCommandLine=|coreos.oem.id=ibmcloud
+ConditionKernelCommandLine=|flatcar.oem.id=ibmcloud
+ConditionKernelCommandLine=|coreos.oem.id=vultr
+ConditionKernelCommandLine=|flatcar.oem.id=vultr
+# Addition:
+ConditionKernelCommandLine=|coreos.oem.id=packet
+ConditionKernelCommandLine=|flatcar.oem.id=packet
```

Comment:
Do we have a supported ID for packet using the new name (equinix metal)?

Reply:
Not yet, both `coreos-base/oem-packet/files/grub.cfg` and `coreos-base/oem-packet/files/oem-release` use `packet`